server.js

```diff
@@ -0,0 +1,121 @@
+const express = require('express');
+
+const server = express();
+
+const knex = require('knex')
+const knexConfig = require('./knexfile')
+
+const db = knex(knexConfig.development)
+const ProjName = require('./projects/project-model.js');
+server.use(express.json());
+
+
+//Retrieve projects
+server.get('/api/projects', (req, res) => {
+	db('projects')
+		.then(projects => {
+			res.json({ data: projects })
+		}).catch(err => {
+			res.status(500).json({ message: 'Failed to get projects' });
+		});
+
+})
+
+//Retrieve resources
+server.get('/api/resources', (req, res) => {
+	db('resources')
+		.then(resources => {
+			res.status(200).json({ data: resources })
+		})
+		.catch(err => {
+			res.status(500).json({ message: 'Failed to get resources' });
+		});
+})
+
+// Post new resource, notes are optional
+server.post('/api/resources', (req, res) => {
+	const resourceName = req.body.name
+	const resourceDesc = req.body.description
+	if (!resourceName || !resourceDesc) {
+		res.status(401).json({ message: 'Please include a name and a description field!' })
+	}
+
+	db('resources')
+		.insert({ name: resourceName, description: resourceDesc })
+		.then(resourceId => res.status(201).json({ data: resourceId })
+
+		).catch(err => {
+			res.status(500).json({ message: 'Failed to post a new resource!' });
+		});
+
+})
+
+
+//Retrieve tasks
+server.get('/api/tasks', (req, res) => {
+	ProjName.findName(db)
+		.then(tasks => {
+			res.status(200).json({ data: tasks })
+		}).catch(err => {
+			console.log(err);
+			res.status(500).json({ message: 'Failed to get tasks.' });
+		});
+
+
+})
+
+// Create a new task, check for description and existing id 
+server.post('/api/tasks', (req, res) => {
+	const taskDes = req.body.description
+	const projectIds = req.body.projectId
+	if (!taskDes) {
+		res.status(400).json({ message: 'Provide a completed task description field!' })
+	} else if (!projectIds) {
+		res.status(400).json({ message: 'Provide a projectId field with a valid projectId number!' })
+	}
+	db('projects')
+		.select()
+		.where({ id: projectIds })
+		.then(project => {
+			if (project.length > 0) {
+				db('tasks')
+					.insert(req.body)
+					.then(projectIds => {
+						ProjName.findName(db)
+							.select()
+							.where({ projectId: projectIds })
+							.then(tasks => {
+								res.status(201).json({ data: tasks })
+							}).catch(err => {
+								res.status(500).json({ message: 'Failed to update task!' });
+							});
+					}).catch(err => {
+						res.status(500).json({ message: 'Failed to post a new resource!' });
+					});
+			} else {
+				res.status(400).json({ message: "That projectId does not exist!" })
+			}
+		})
```

Comment:
There are (more than, but I'll focus on) two different ways you could type this query.

One simple way would be- if we add a task correctly we **only return the id:**
```js
db('projects')
    .select()
    .where({ id: projectIds })
    .then(project => {
        if (project.length > 0) {
            db('tasks')
                .insert(req.body)
                .then(ids => {
                    res.status(201).json({ message: `Added task with id ${ids[0]}.` })
                }).catch(err => {
                    res.status(500).json({ message: 'Failed to post a new resource!' });
                });
        } else {
            res.status(400).json({ message: "That projectId does not exist!" })
        }
    }).catch(err => {
        res.status(500).json({ message: 'Failed to post a new resource!' });
    });
```
We don't have to query the database an extra time to get extra task information, and we can just return the task id 👍 

Now, I think you're trying to return **all of the tasks for a given project** after inserting. If that's the case, I'd write the code like so:
```js
db('projects')
    .select()
    .where({ id: projectIds })
    .then(project => {
        if (project.length > 0) {
            db('tasks')
                .insert(req.body)
                .then(taskId => { // Typing taskId instead of projectIds . . .
                    ProjName.findName(db)
                        .select()
                        .where({ projectId: projectIds }) // . . . So here, we're still referencing the projectIds from our req.body
                        .then(tasks => {
                            res.status(201).json({ data: tasks })
                        }).catch(err => {
                            res.status(500).json({ message: 'Failed to update task!' });
                        });
                }).catch(err => {
                    res.status(500).json({ message: 'Failed to post a new resource!' });
                });
        } else {
            res.status(400).json({ message: "That projectId does not exist!" })
        }
    }).catch(err => { // Always have this final catch statement! If the initial projects throws an error, we need a catch here '' '
        res.status(500).json({ message: 'Failed to post a new resource!' });
    });
```

There was one line you had that was bugging the code so it didn't correctly return all of the tasks. Once that was fixed, we just need the final catch at the very bottom of the code to catch any final errors.